packages/wdio-utils/src/envDetector.ts

```diff
@@ -200,6 +200,36 @@ function isAppiumCapability(capabilityName: string) {
     return capabilityName.startsWith(APPIUM_CAPABILITY_PREFIX)
 }
 
+/**
+ * Detects if the session is in a native context
+ * @param  {object}  capabilities session capabilities
+ * @return {Boolean}              true if session is in a native context
+ */
+function isNativeContext(capabilities: WebdriverIO.Capabilities): boolean {
+    const isAppiumAppCapPresent = (capabilities: Capabilities.RequestedStandaloneCapabilities) => {
+        const appiumKeys = ['app', 'bundleId', 'appPackage', 'appActivity', 'appWaitActivity', 'appWaitPackage']
+        return appiumKeys.some(key => (capabilities as Capabilities.AppiumCapabilities)[key as keyof Capabilities.AppiumCapabilities] !== undefined)
+    }
+    const isBrowserNameFalse = !!capabilities.browserName === false
+    // @ts-expect-error
+    const isAutoWebviewFalse = capabilities.autoWebview !== true
+
+    return isBrowserNameFalse && isAppiumAppCapPresent(capabilities) && isAutoWebviewFalse
+}
+
+/**
+ * Returns the mobile context
+ * @param {object}  capabilities session capabilities
+ * @returns {string | undefined}    The mobile context
+ */
+function mobileContext(capabilities: WebdriverIO.Capabilities): string | undefined {
+    return isNativeContext(capabilities) ? 'NATIVE_APP' :
+        // Android webviews are always WEBVIEW_<package_name>, Chrome will always be CHROMIUM
+        // We can only determine it for Android and Chrome, for all other, including iOS, we return undefined
+        isAndroid(capabilities) && capabilities?.browserName?.toLowerCase() === 'chrome' ? 'CHROMIUM' :
+            undefined
+}
```

Comment:
And move this into the `ContextManager` class to compute the initial value.

Reply:
👍